### PR TITLE
Title bars

### PR DESCRIPTION
**Pull request by louismaximepiton**
Previews : 
https://deploy-preview-956--boosted.netlify.app/docs/5.1/examples/title-bars/
https://deploy-preview-956--boosted.netlify.app/docs/5.1/components/title-bars/

### Development

- [x] Should match specs (eg. either the Web UI Kit or any pattern from the WAI — or both…)
- [x] Docs added:
  - including the "Sass" part using `scss-docs` shortcode
  - in /about/overview/#custom-components if it is a new Orange custom component
  - in /getting-started/introduction/#components if it is a new Orange custom component that requires JavaScript (and Popper)
  - in /customize/overview#csps-and-embedded-svgs if it is a new Orange custom component that includes embedded SVGs in our CSS
  - in /forms/validation/?#supported-elements if it is a new Orange custom component that is a form control
  - in /forms/overview/ if it is a new Orange custom component that is a form control
- [x] Check (and fix) RTL version
- [x] Run linters
- [x] Run compilers
- [x] Tests added for JS-side
- [x] Run tests
- [ ] Manually run BrowserStack test
- [ ] Manually run Percy test
- [x] Cross-browser test:
  - Firefox ESR
  - Latest Edge, Chrome, Firefox, Safari
  - iOS Safari
  - Chrome & Firefox on Android
- [ ] Clean up the branch using `rebase -i`
- [ ] Commited with `feat(…): …` message
- [ ] Mention it in Migration Guide (if `back-from-v4`): renamed variables, changes in markup requirement,  etc.

### Reviews

- [x] Code review
- [x] A11y review
- [x] Design review

### CODE REVIEW

**Comment by julien-deramond** [comment]
:warning: There are 7 pa11y errors of type `Error: Document must have one main landmark (https://dequeuniversity.com/rules/axe/3.5/landmark-one-main?application=axeAPI)`.
Try to run it locally:
```
$ npm run docs-accessibility
$ npm run docs-pa11y-html
```
And open `.pa11y/index.html` to see the results.

2 possibilities here:
* it is possible to fix the issue
* the issue is "normal" and you'll need to hide it in `build/.pa11ci.json` (and reference it in the Wiki when it'll be merged)

**Comment by julien-deramond** [comment]
Note: If https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/982 is closed before the review of this PR, we must apply here the Title Bars to the documentation templates. Otherwise, we need to modify the description of the issue to notify the developer to use Title Bars.

**Comment by louismaximepiton** [comment]
On hold while waiting for a solution for the padding-x.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:181]: Remove this part. It is too complicated to maintain.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:187]: Remove this part. It is too complicated to maintain.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:64]: Are we sure we can use all supporting colors? It is not specified in the [Design Guidelines](https://system.design.orange.com/0c1af118d/p/774477-local-headers/b/743cd0) where we can only see white/black/green background.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:50]: Remove this part difficult to maintain.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:44]: Remove this part difficult to maintain.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:9]: I don't understand why there is an "Examples" part an then a "Title bars" with almost the same content. IMO it should be merged into one: "Title bars".

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:4]: Remove the reference to breadcrumbs and local navigation bars since it is not yet implemented.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/orange-local-headers.md:112]: Instead of writing title representing the current section (e.g. "Srcset") I would stick with the "Title" string.

File: scss/_title-bars.scss
Comment (julien-deramond) [inline scss/_title-bars.scss:47]: * It seems that there is a 1px difference (vertical alignment) with the [design guidelines](https://system.design.orange.com/0c1af118d/p/774477-local-headers/b/743cd0/i/66611057) in all modes. I think the title should be lower by 1px :thinking: → I observe the same issue with the vertical alignment of the image.
* From 480px (equal and below), it seems that the title is 1px too on the right. I think the title should be more to the left of 1px :thinking:

File: scss/_title-bars.scss
Comment (julien-deramond) [inline scss/_title-bars.scss:48]: The 3 comments with the corresponding px sizes can be removed.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
First series of comments without digging at the examples and .scss.

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (louismaximepiton) [reply site/content/docs/5.1/components/orange-local-headers.md:64]: To discuss with designers, I'll contact them about that

File: scss/_title-bars.scss
Comment (louismaximepiton) [reply scss/_title-bars.scss:47]: The image's height is a ratio of Zeplin's image height and font-size (which is around 1.8). As this is not pixel perfect, do you think that an harcoded value would be better ?

File: site/content/docs/5.1/components/orange-local-headers.md
Comment (louismaximepiton) [reply site/content/docs/5.1/components/orange-local-headers.md]: In this case, it wraps because of the width of the main container, do you think that I should add : `word-break: keep-all` ?

**Comment by julien-deramond** [comment]
@louismaximepiton https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1009 is merged so we should try to include Title bars into "Examples" and "Guidelines" pages for the titles.

**Review by isabellechanclou — APPROVED** [review]

**Comment by Franco-Riccitelli** [comment]
I have completed a design review of the title bars and there are no issues - great work!
Please note, title bars can be displayed with black or white backgrounds. When using black or white backgrounds, illustrations can be included as an option, but it's not necessary.
Supporting colours can also be used as the background colour for title bars, but not the supporting light and dark tint colours. When using one of the supporting colours, Illustrations must be used.
The illustration used in the implementation is a good example of how to use illustrations in title bars, especially across the breakpoints, but it must be possible to replace this with other illustrations if needed. Thanks

**Comment by isabellechanclou** [comment]
Thanks @Franco-Riccitelli for your review and comments. Considering the rules you mentioned, we added them to complete the documentation part accordingly. By the way, it would be useful to have theses rules also mentioned in ODS.

**Comment by louismaximepiton** [comment]
FYI, Ok for the accessibility team.

**Comment by julien-deramond** [comment]
Added a CSS section in the documentation via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/956/commits/d02f9f004fba06fe0570556315ceeacc475677e6

File: scss/_title-bars.scss
Comment (julien-deramond) [inline scss/_title-bars.scss:4]: Don't we rather need to create intermediate `$title-bar-*` Scss vars instead of using directly `$h2-font-size`, `$h2-spacing` and `$gray-500`?

File: scss/_title-bars.scss
Comment (julien-deramond) [inline scss/_title-bars.scss:13]: Apparently in the DSM we have to following sizes for the header: 60px 50px 30px. It looks like [`.display-1`](https://boosted.orange.com/docs/5.1/content/typography/#display-headings). Can't we use this class instead?

File: site/content/docs/5.1/components/title-bars.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/title-bars.md:11]: Global header doesn't exist in Boosted doc. Maybe talk about (+ link) Orange navbar here.

File: site/data/examples.yml
Comment (julien-deramond) [inline site/data/examples.yml:21]: We probably should remove the "Components" category and move its content in this one. Bootstrap doesn't have "Components".

**Review by julien-deramond — CHANGES_REQUESTED** [review]
- Some space's missing between the border and the description in mobile mode.

Screenshot from 2022-05-03 08-25-54

- The example thumbnail should be update to show the content of the examples rather than an image contained in the title bars.

File: scss/_title-bars.scss
Comment (julien-deramond) [reply scss/_title-bars.scss:13]: OK I understand, the breakpoints are different between the DSM for this component and our breakpoints in Boosted regarding headers... Let's add `.display-1` in this PR and fix the behavior in the CSS, and then create an issue to align the DSM and Boosted for the headers.

Comment (louismaximepiton) [reply scss/_title-bars.scss:13]: Just a reminder for the one that will create the issue, the actual breakpoints for `display-*` are `sm` and `lg`. Here, we need `md` and `xl`.

File: site/content/docs/5.1/examples/title-bars/index.html
Comment (julien-deramond) [inline site/content/docs/5.1/examples/title-bars/index.html:15]: IMO this text can be improved to explain that all three examples are not using the same "image system" and that only the blue one (the 2nd) needs an extra class named `.example-title-bar-custom-background` and an extra CSS definition.

File: site/content/docs/5.1/components/title-bars.md
Comment (julien-deramond) [inline site/content/docs/5.1/components/title-bars.md:41]: `.mt-3` is enough, I think you can remove those `.bg-body`s everywhere when used for separators.

**Review by julien-deramond — APPROVED** [review]
Final review with 2 small comments. After the fixes, let's merge :rocket:

**Pull request merged**
